docs/java-rest/high-level/dataframe/put_data_frame.asciidoc

```diff
@@ -0,0 +1,93 @@
+--
+:api: put-data-frame-transform
+:request: PutDataFrameTransformRequest
+:response: AcknowledgedResponse
+--
+[id="{upid}-{api}"]
+=== Put Data Frame Transform API
+
+The Put Data Frame Transform API is used to create a new {dataframe-job}.
+
+The API accepts a +{request}+ object as a request and returns a +{response}+.
+
+[id="{upid}-{api}-request"]
+==== Put Data Frame Request
+
+A +{request}+ requires the following argument:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-request]
+--------------------------------------------------
+<1> The configuration of the {dataframe-job} to create
+
+[id="{upid}-{api}-config"]
+==== Data Frame Transform Configuration
+
+The `DataFrameTransformConfig` object contains all the details about the {dataframe-job}
+configuration and contains the following arguments:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-config]
+--------------------------------------------------
+<1> The data frame transform ID
+<2> The source index or index pattern
+<3> The destination index
+<4> Optionally a QueryConfig
+<5> The PivotConfig
+
+[id="{upid}-{api}-query-config"]
+==== QueryConfig
+
+The query with which to select data from the source index.
+If not set a `match_all` query is used by default.
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-query-config]
+--------------------------------------------------
+
+==== PivotConfig
+
+Defines the pivot transform `group by` fields and the aggregation to reduce the data.
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-pivot-config]
+--------------------------------------------------
+
+===== GroupConfig
+The grouping terms. Defines the group by and destination fields
+which are produced by the grouping transform. There are 3 types of
```

Comment:
`grouping transform`, you mean `pivot transform` or as I suggested above `pivot function`